### ISSUE TITLE
SDK should be updated to support secure connections

### ISSUE THREAD

**Issue by kurtzettel** [splunk/splunk-sdk-java#72]
Due to Poodle most security conscious organizations have disable SSLv3 on many of their servers.  Java has also disabled SSLv3 in several releases.  While some people have recommended commenting out: jdk.tls.disabledAlgorithms=SSLv3 it would be better if the sdk supported TLS.  There are several forks that have changed this.

Related links:
http://answers.splunk.com/answers/210201/i-changed-splunk-from-using-sslv3-to-tlsv12-and-th.html
http://answers.splunk.com/answers/209379/no-appropriate-protocol-protocol-is-disabled-or-ci.html

**Comment by shakeelmohamed** [comment]
Hi @kurtzettel,

We are aware of this issue, and will look into it. We're always open to a pull request!

Thanks

**Comment by dojadop** [comment]
:+1: We just upgraded one environment to use java 1.8.0_40 and started running into this issue

Does anyone know the impact to older versions of java if this was done suggested in this post http://answers.splunk.com/answers/210201/i-changed-splunk-from-using-sslv3-to-tlsv12-and-th.html:

```
 SSLContext context = SSLContext.getInstance("SSL");
 to
 SSLContext context = SSLContext.getInstance("TLSv1.2");
 
 public static final String[] PROTOCOLS = {"SSLv3"};
 to
 public static final String[] PROTOCOLS = {"TLSv1.2"};
```

If there isn't seems like a fairly trivial solution.

**Comment by dojadop** [comment]
Nevermind i just found this https://github.com/splunk/splunk-sdk-java/pull/61

Seems like there is an issue and would require a different implementation.  @kurtzettel did your solution work in Java 8?

**Comment by shakeelmohamed** [comment]
Hi @dojadop @kurtzettel,

We have a potential fix in #75, please take a look

**Comment by dojadop** [comment]
Thanks!  Sorry for the slow response @shakeelmohamed.  It is working for me.

**Comment by shakeelmohamed** [comment]
Hi @dojadop @kurtzettel,

We've just released v1.4.0, adding some functionality to allow you to specify which protocol you want to use when connecting to Splunk.
The [`ssl_protocols`](https://github.com/splunk/splunk-sdk-java/blob/release/1.4.0/examples/com/splunk/examples/ssl_protocols/Program.java) example will show you a couple different ways to work with Java 8 and/or TLS.

Once again, thanks for your patience!
